### ISSUE TITLE
core add-on should be accessible [a11y]

### ISSUE THREAD

**Issue by rhelmer** [mozilla-rally/core-addon#270]
Svelte already seems to check for many common things, but Mozilla has its own [accessibility guidelines](https://wiki.mozilla.org/Accessibility/Guidelines) that we should check against.

See also https://github.com/sveltejs/svelte/issues/820 which mentions an eslint plugin that we should look into as well.

We should also ask the Mozilla a11y folks for a reviews and advice on this as we go. We'd like to make everything from the website to the core and study add-ons accessible.

**Comment by rhelmer** [comment]
> Svelte already seems to check for many common things, but Mozilla has its own [accessibility guidelines](https://wiki.mozilla.org/Accessibility/Guidelines) that we should check against.

There is also this review checklist:
https://searchfox.org/mozilla-central/source/docs/bug-mgmt/processes/accessibility-review.md

Asa says that we can do this outside of Bugzilla as well.

**Comment by rhelmer** [comment]
[pa11y](https://github.com/pa11y/pa11y) looks interesting, it uses Puppeteer and Chromium to check against an `http:` or `file:` URL. Here it is running against the latest [Rally site](github.com/mozilla-rally/site/):

```zsh
% pa11y http://localhost:8080

Welcome to Pa11y

 > Running Pa11y on URL http://localhost:8080

Results for URL: http://localhost:8080/

 • Error: Heading tag found with no content. Text that is not intended as a heading should not be marked up with heading tags.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.H42.2
   ├── html > body > header > div > div:nth-child(1) > div > h1
   └── <h1> <a class="Header-log...</h1>

 • Error: This button element does not have a name available to an accessibility API. Valid names are: title undefined, element content, aria-label undefined, aria-labelledby undefined.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Button.Name
   ├── html > body > header > div > div:nth-child(1) > button
   └── <button class="Header-navToggle btn no-js-hide" type="button" data-expands="nav" data-expands-height=""> <span class="Header-na...</button>

 • Error: This link points to a named anchor "FIXME" within the document, but no anchor exists with that name.
   ├── WCAG2AA.Principle2.Guideline2_4.2_4_1.G1,G123,G124.NoSuchID
   ├── html > body > header > div > div:nth-child(3) > a:nth-child(3)
   └── <a class="mzp-c-button mzp-t-md hide-addons" href="#FIXME">Get Mozilla Rally</a>

 • Error: This link points to a named anchor "FIXME" within the document, but no anchor exists with that name.
   ├── WCAG2AA.Principle2.Guideline2_4.2_4_1.G1,G123,G124.NoSuchID
   ├── html > body > main > div:nth-child(4) > div:nth-child(2) > div > div:nth-child(1) > div > div > div > a:nth-child(3)
   └── <a class="ReadyRally-cta mzp-c-button mzp-t-md hide-addons" href="#FIXME">Get Mozilla Rally</a>

 • Error: Anchor element found with a valid href attribute, but no link content has been supplied.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.A.NoContent
   ├── html > body > footer > div > div:nth-child(1) > a:nth-child(1)
   └── <a class="Footer-logo" href="/"> <svg xmlns="http://www...</a>

 • Error: This link points to a named anchor "FIXME" within the document, but no anchor exists with that name.
   ├── WCAG2AA.Principle2.Guideline2_4.2_4_1.G1,G123,G124.NoSuchID
   ├── html > body > footer > div > div:nth-child(1) > a:nth-child(4)
   └── <a class="Footer-cta mzp-c-button mzp-t-secondary mzp-t-dark hide-addons" href="#FIXME">Get Mozilla Rally</a>

6 Errors
```

We should consider if it's better to run this standalone, or try to integrate this feature into our existing Selenium tests.

**Comment by Dexterp37** [comment]
I think we should probably add another CI step to run this. It seems a good idea to have such a tool catch problems early. Good catch!

**Comment by marniepw** [comment]
Post-alpha, can reach back out to Asa for a thorough review.
